frontend/src/components/AddConcepts.tsx

```diff
@@ -0,0 +1,395 @@
+  return <ConceptStringSearch />;
+
+}
+// from https://stackoverflow.com/a/66167322/1368860
+function ConceptStringSearch() {
+  const dataGetter: DataGetter = useDataGetter();
+  const [codeset_ids, codesetIdsDispatch] = useCodesetIds();
+  const [cids, cidsDispatch] = useCids();
+  const [searchText, setSearchText] = React.useState("");
+  const c: number[] = [];
+  const [have_concept_ids, setHaveConceptIds] = React.useState(c);
+  const [found_concept_ids, setFoundConceptIds] = React.useState(c);
+  const lastRequest = React.useRef(null);
+  const windowSize = useWindowSize();
+
+  // this effect will be fired every time searchText changes
+  React.useEffect(() => {
+    (async () => {
+      // setting min lenght for searchText
+      if (searchText.length >= 3) {
+        // updating the ref variable with the current searchText
+        lastRequest.current = searchText;
+        console.log('before search fetch');
+        const r = await dataGetter.fetchAndCacheItems(dataGetter.apiCalls.concept_search, searchText)
+        console.log('after search fetch');
+        if (lastRequest.current === searchText) {
+          // console.log("response is valid!", r);
+          setFoundConceptIds(r);
+        } else {
+          // console.log("discarding api response", searchText, lastRequest.current);
+        }
+      }
+    })();
+  }, [searchText]);
+
+  React.useEffect(() => {
+    (async () => {
+      // csmi for these codeset_ids should already be cached
+      console.log('before csmi fetch');
+      const csmi: { [key: number]: Concept } = await dataGetter.fetchAndCacheItems(dataGetter.apiCalls.cset_members_items, codeset_ids);
+      console.log('after csmi fetch');
+      let h = uniq(flatten(
+          Object.values(csmi).map(d => Object.values(d))
+      ).filter(d => d.item).map(d => d.concept_id));
+      // let h = Object.values(csmi).map(c => c.concept_id);
+      h = setOp('union', h, cids);
+      setHaveConceptIds(h);
+    })();
+  }, [codeset_ids, cids]);
+
+  const paddingLeft = 100, paddingRight = 100;
+  const padding = paddingLeft + paddingRight;
+  const divWidth = Math.min(windowSize[0], 1300) - padding;
+  const displayConceptIds = setOp('difference', found_concept_ids, have_concept_ids);
+  const hiddenMatches = setOp('intersection', found_concept_ids, have_concept_ids);
+  console.log('render 1');
+  return (
+    <div style={{paddingLeft, paddingRight, width: divWidth}}>
+      <h1>Concept Search</h1>
+      <input style={{width: 350}} type="text" placeholder="match characters in concept name"
+             onChange={(e) => setSearchText(e.target.value)}
+             value={searchText}
+             autoFocus={true}
+      />
+      {'\u00A0'}{'\u00A0'}{'\u00A0'}{found_concept_ids.length ? found_concept_ids.length.toLocaleString() + ` concepts match "${searchText}"` : ""}
+      ;{'\u00A0'}{hiddenMatches.length ? hiddenMatches.length.toLocaleString() + ' already included and not listed' : ""}
```

Comment:
```suggestion
      {'\u00A0'}{hiddenMatches.length ? hiddenMatches.length.toLocaleString() + ' already included and not listed' : ""}
```
I think this semicolon here is manifesting in:
<img width="473" alt="Screenshot 2024-05-29 at 4 38 06 PM" src="https://github.com/jhu-bids/TermHub/assets/13045020/49ef37e5-7a72-4265-a466-7bdfd505d33d">